### PR TITLE
Applying govuk-lint (just app and spec folders)

### PR DESCRIPTION
**Pull request by fjavierm**

### CODE REVIEW

**Comment by gidsg** [comment]
should be also run it against `/spec`?

**Comment by gidsg** [comment]
If we are doing this we should also set it up
1) as a pre-commit hook
2) as a check in travis CI

else it will just slowly rot again.

**Comment by arnau** [comment]
I agree with the check in Travis.

File: app/mailers/devise_mailer.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class DeviseMailer < Devise::Mailer
   helper :application # gives access to all helpers defined within `application_helper`.
   include Devise::Controllers::UrlHelpers # Optional. eg. `confirmation_url`
 
-  def reset_password_instructions(record, token, opts={})
+  def reset_password_instructions(record, token, _opts = {})
```
Comment (arnau) [inline app/mailers/devise_mailer.rb]: If `opts` is not used at all, either remove it entirely or, if the expected signature requires it explicitly, don't assign a default.

File: app/models/concerns/multi_step_model.rb [hunk 2]
```diff
@@ -46,4 +46,7 @@ def method_missing(method_name, *args, &block)
     end
   end
 
-end
\ No newline at end of file
+  def respond_to_missing?(method, *)
+    method =~ /^step(\d+)\?$/ || super
+  end
```
Comment (arnau) [inline app/models/concerns/multi_step_model.rb]: Is this method new? if so, I don't think it should be part of this PR. In any case, I'm interested in knowing the rationale behind this :)

File: app/permissions/user_policy.rb [hunk 3]
```diff
@@ -18,7 +16,7 @@ def new?(current_user, role, team_id)
       return false unless current_user.present?
 
       if current_user.admin?
-        if role.present? then
+        if role.present?
           return true
         else
           return false
```
Comment (arnau) [inline app/permissions/user_policy.rb]: Isn't that just `return role.present?`. Something like:

```ruby
return role.present? if current_user.admin?
```

File: app/models/concerns/multi_step_model.rb (hunk 2 above)
Comment (fjavierm) [reply app/models/concerns/multi_step_model.rb]: It's just a recommended practice when you use the method `method_missing` that we were using.

File: app/mailers/devise_mailer.rb (hunk 1 above)
Comment (fjavierm) [reply app/mailers/devise_mailer.rb]: the signature in the parent class is `reset_password_instructions(record, token, opts={})`. Just the same signature

**Comment by fjavierm** [comment]
@arnau @gidsg Can you take another quick look? I've added the `spec` folder files too and addressed some of the comments.

**Comment by gidsg** [comment]
@fjavierm LGTM. Will be happy to approve if you can do another commit adding same check in Travis CI. I think you can do something like this: https://github.com/alphagov/gsd-view-data/blob/399bb876820dae8e011c45a9c1a2a4787a00a666/.travis.yml#L38

**Pull request merged**